flink-java/src/main/java/org/apache/flink/api/java/Utils.java

```diff
@@ -138,6 +138,12 @@ public void writeRecord(T record) throws IOException {
 
 		@Override
 		public void close() {
+			// when the sink is up but not initialized and the job fails due to other operators,
+			// it is possible that close() is called when open() is not called,
+			// so we have to do this null check
+			if (accumulator == null) {
+				accumulator = new SerializedListAccumulator<>();
```

Comment:
Why not just skipt adding this accumulator into context?

Reply:
The user of this `CollectHelper` expects this accumulator to appear in the result. Not adding this accumulator will result in additional checks on the user side.

Reply:
But the case is job failed, I think it's hard to make sure the accumulator is in result even if job is failed. What if the operator/task/container is never aroused?